BudgetTracker.BudgetSquirrel.Web/Application/Auth/UserService.cs

```diff
@@ -0,0 +1,49 @@
+using BudgetTracker.Business.Auth;
+using BudgetTracker.Business.Budgeting;
+using BudgetTracker.Business.Ports.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace BudgetTracker.BudgetSquirrel.Application.Auth
+{
+    public class UserService
+    {
+        private BudgetCreation _budgetCreator;
+        private IUserRepository _userRepository;
+
+        public UserService(BudgetCreation budgetCreator, IUserRepository userRepository)
+        {
+            _budgetCreator = budgetCreator;
+            _userRepository = userRepository;
+        }
+
+        public async Task<(User, Budget)> RegisterUserAndRootBudget(CreateUserViewModel setupInput)
+        {
+            EnsurePasswordConfirmation(setupInput.Password, setupInput.ConfirmPassword);
```

Comment:
I usually like to `this.` for private methods so it easier to see where they came from.